src/main/java/com/mucommander/main/AutoProcessor.java

```diff
@@ -74,9 +89,17 @@ public class AutoProcessor
     **/
     public static void process(Map configMap, BundleContext context)
     {
-        configMap = (configMap == null) ? new HashMap() : configMap;
-        processAutoDeploy(configMap, context);
-        processAutoProperties(configMap, context);
+        ExecutorService executor = null;
+        try {
+            executor = Executors.newFixedThreadPool(8);
```

Comment:
does it need to be within the try-finally block?..

Reply:
I would prefer to have it in finally, if sth goes wrong we would like to finish jobs in executor before going anymore further. Yeah, I know that muC will be in strange condition, but at least more tasks will be finished and maybe a better log will be available.

Reply:
ok, but even in a scenario that is unlikely to happen in which we get runtime exception from the call to newFixedThreadPool, there won't be job submitted yet, right?

Reply:
I think that `finally` is kinda a good practice that I would keep it.

Reply:
ok